src/lib/storage/value_column.cpp

```diff
@@ -84,11 +92,26 @@ void ValueColumn<T>::append(const AllTypeVariant& val) {
     return;
   }
 
-  Assert(!is_null, "ValueColumns is not nullable but value passed is null.");
+  Assert(!is_null, "ValueColumn is not nullable but value passed is null.");
 
   _values.push_back(type_cast<T>(val));
 }
 
+template <typename T>
+void ValueColumn<T>::append_typed_value(const std::optional<T> value_or_null) {
```

Comment:
Do you really need this, or can you just pre-build the null/value vectors in Sort? I am really against needlessly extending the interface of our core storage classes.

Reply:
Did you push? Sort still seems to use `append_typed_value` as far as I can tell.

Reply:
Yes, see [here](https://github.com/hyrise/hyrise/commit/e5f8a8e9a5c045fe640460e37ce762c90227c939#diff-a735884cb8be5955a5069458e2908620L112)